app/src/main/java/me/saket/dank/ui/preferences/PreferenceGroupsScreen.kt

```diff
@@ -85,7 +85,7 @@ class PreferenceGroupsScreen(context: Context, attrs: AttributeSet) :
   override fun onSaveInstanceState(): Parcelable? {
     val values = Bundle()
 
-    groupChanges.value.ifPresent { group ->
+    groupChanges.value?.ifPresent { group ->
```

Comment:
This looks a bit unusual. Did you check whether we could get rid of the use of `Optional` here? The requirement to use null check suggests that it should be possible.

Reply:
Not yet, I'll get right on that.

Reply:
https://github.com/Tunous/Dawn/pull/85/files/c36094af4022f19eebbcb2046abc03061587b32d#diff-240fec58b3237ecd66e74db91fb096e9R214 does not accept null values so it's necessary to wrap it with an `Optional` :(